crates/infra/cli/src/commands/publish/cargo/mod.rs

```diff
@@ -77,14 +79,16 @@ fn update_cargo_lock() -> Result<()> {
         .run()
 }
 
-fn run_cargo_publish() -> Result<()> {
+fn run_cargo_publish(dry_run: DryRun) -> Result<()> {
     let mut command = Command::new("cargo")
         .arg("publish")
         .property("--package", USER_FACING_CRATE)
         .flag("--all-features");
 
-    if !GitHub::is_running_in_ci() {
-        println!("Attempting a dry run, since we are not running in CI.");
+    if dry_run.is_yes() || !GitHub::is_running_in_ci() {
```

Comment:
> Attempting a dry run, since we are not running in CI or a dry run was requested

With the new `--dry-run` flag, I wonder if we should just rely on it, instead of checking `is_running_in_ci()`, as it would serve the same purpose.

Reply:
Sounds good, but then the local default would be to always publish (as `--dry-run` is not specified by default). Makes sense as that's the way most publishing CLIs work but it would be good to have an additional prompt (Are you sure? You're going to publish XYZ) before doing so.

In any case, I'm happy with pruning the dry-run from this PR as I'm only interested in landing the fix quickly; we can brainstorm the design later, if needed.

Reply:
I'm not worried about accidentally publishing from a local machine, since developer accounts don't even have access permissions.
But just a nit. Not blocking of course.

Reply:
> I was paranoid and I wanted a way to explicitly do a dry run in CI, even if we did not provide tokens in my testing CI run.

Sounds good.